drivers/hmis/spec/system/hmis/form_builder_spec.rb

```diff
@@ -0,0 +1,257 @@
+require 'rails_helper'
+require_relative '../../requests/hmis/login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+# Custom matcher for asserting order of text in Capybara page body, helpful for testing reordering of form items.
+# https://stackoverflow.com/questions/64002940/is-it-possible-to-find-out-if-an-element-comes-before-the-second-element-in-capy
+RSpec::Matchers.define :appear_before do |later_content|
+  match do |earlier_content|
+    page.body.index(earlier_content) < page.body.index(later_content)
+  end
+end
+
+RSpec.feature 'HMIS Form Builder', type: :system do
+  include_context 'hmis base setup'
+
+  let!(:ds1) { create(:hmis_data_source, hmis: 'localhost') }
+  let!(:access_control) { create_access_control(hmis_user, p1) }
+  let!(:published) { create :custom_assessment_with_custom_fields_and_rules, title: 'System Test 1', identifier: 'system_test_1', data_source: ds1 }
+  let!(:draft) { create :custom_assessment_with_custom_fields_and_rules, title: 'System Test 2', identifier: 'system_test_2', status: 'draft', data_source: ds1 }
+  let!(:advanced_draft) { create :custom_assessment_with_field_rules_and_autofill, identifier: 'advanced_form', status: 'draft', data_source: ds1 }
+
+  context 'Form builder happy path' do
```

Comment:
I woud suggest encapsulating the fixtures inside `context`s, so the tests are easier to read. That way, there are only 1 or 2 fixtures at play for a given test, and we don't need to worry about mapping back the identifiers like "system_test_1" vs "system_test_2" etc.


For example:

```
context 'with no existing form' do
   it 'creates a new draft' do
   end
end

context 'with an existing published form' do
   let!(:published) { create :custom_assessment_with_custom_fields_and_rules, data_source: ds1 }
     before(:each) do
        visit "/admin/forms/#{published.identifier}"
      end
   it 'creates a new draft' do
   end
end

context 'with an existing draft form' do
   let!(:published) { create :custom_assessment_with_custom_fields_and_rules, status: 'published', data_source: ds1 }
   let!(:draft) { create :custom_assessment_with_custom_fields_and_rules, identifier: published.identifier, status: 'draft', data_source: ds1 }

     before(:each) do
        visit "/admin/forms/#{draft.identifier}"
      end

   it 'does not create a new draft' do
   end

   it 'publishes the draft and retires the old form' do
   end
end
```